commit 5e1284136211e73df6468842e6130cce2ef94e23
Author: Matus Kral <matuskral@me.com>
Date:   Wed Aug 04 00:35:47 2014 +0200

  this fixes race condition upon CEC init making XBMC become active source very hard

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 8919380..23dfaa3 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -332,8 +337,6 @@ bool CPeripheralCecAdapter::OpenConnection(void)
       if (!bConnectionFailedDisplayed)
         CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Error, g_localizeStrings.Get(36000), g_localizeStrings.Get(36012));
       bConnectionFailedDisplayed = true;
-
-      Sleep(10000);
     }
   }
 
   m_queryThread->StopThread(true);
@@ -1536,6 +1551,17 @@ bool CPeripheralCecAdapterUpdateThread::SetInitialConfiguration(void)
   if (!WaitReady())
     return false;
 
+  bool bContinue(false);
+  if (m_configuration.bActivateSource == 1)
+  {
+    while (!m_adapter->m_cecAdapter->IsLibCECActiveSource() && bContinue)
+      bContinue = !m_event.WaitMSec(1000);
+
+    CLog::Log(LOGDEBUG, "%s - XBMC is now activesource", __FUNCTION__);
+  }
+
+  m_adapter->m_bIsReady = true;
+
   UpdateMenuLanguage();
 
   // request the OSD name of the TV
@@ -1547,8 +1573,6 @@ bool CPeripheralCecAdapterUpdateThread::SetInitialConfiguration(void)
   if (!strAmpName.empty())
     strNotification += StringUtils::Format("- %s", strAmpName.c_str());
 
-  m_adapter->m_bIsReady = true;
-
   // and let the gui know that we're done
   CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(36000), strNotification);
 
